File: app/src/main/java/com/example/wanininterviewassignment/ChangePassword.java
package com.example.wanininterviewassignment;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.auth.AuthCredential;
import com.google.firebase.auth.EmailAuthProvider;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class ChangePassword extends AppCompatActivity {

    EditText new_password, type_again;
    Button submit;
    FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_change_password);
        
        new_password = findViewById(R.id.new_password);
        type_again = findViewById(R.id.type_again);
        submit = findViewById(R.id.submit);

        Bundle bundle = getIntent().getExtras();
        String email = bundle.getString("email");
        String password = bundle.getString("password");

        mAuth = FirebaseAuth.getInstance();

        submit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(new_password.getText().toString().equals(type_again.getText().toString())){
                    FirebaseUser user = mAuth.getCurrentUser();
                    AuthCredential authCredential = EmailAuthProvider.getCredential(user.getEmail(), password);
                    user.reauthenticate(authCredential).addOnSuccessListener(new OnSuccessListener<Void>() {
                        @Override
                        public void onSuccess(Void unused) {
                            user.updatePassword(new_password.getText().toString()).addOnSuccessListener(new OnSuccessListener<Void>() {
                                @Override
                                public void onSuccess(Void unused) {
                                    Toast.makeText(ChangePassword.this, "Password Updated...", Toast.LENGTH_SHORT).show();
                                    Intent intent = new Intent();
                                    intent.setClass(ChangePassword.this, MainActivity.class);
                                    startActivity(intent);
                                }
                            }).addOnFailureListener(new OnFailureListener() {
                                @Override
                                public void onFailure(@NonNull Exception e) {
                                    Toast.makeText(ChangePassword.this, e.getMessage(), Toast.LENGTH_SHORT).show();
                                }
                            });
                        }
                    });
                }
                else{
                    Toast.makeText(ChangePassword.this, "Input Error...", Toast.LENGTH_SHORT).show();
                }

            }
        });
    }
}

File: app/src/main/java/com/example/wanininterviewassignment/ChangePassword.kt
package com.example.wanininterviewassignment

import androidx.appcompat.app.AppCompatActivity
import android.widget.EditText
import com.google.firebase.auth.FirebaseAuth
import android.os.Bundle
import com.example.wanininterviewassignment.R
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.auth.AuthCredential
import com.google.firebase.auth.EmailAuthProvider
import com.google.android.gms.tasks.OnSuccessListener
import android.widget.Toast
import android.content.Intent
import android.view.View
import android.widget.Button
import com.example.wanininterviewassignment.MainActivity
import com.google.android.gms.tasks.OnFailureListener

class ChangePassword : AppCompatActivity() {
    var new_password: EditText? = null
    var type_again: EditText? = null
    var submit: Button? = null
    var mAuth = FirebaseAuth.getInstance()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_change_password)

        new_password = findViewById(R.id.new_password)
        type_again = findViewById(R.id.type_again)
        submit = findViewById(R.id.submit)

        val bundle = intent.extras
        val email = bundle?.getString("email")
        val password = bundle?.getString("password")

        submit?.setOnClickListener(View.OnClickListener {
            if(new_password?.text.toString() != "" && type_again?.text.toString() != ""){
                if (new_password?.text.toString() == type_again?.text.toString()) {
                    val user = mAuth.currentUser
                    val authCredential = EmailAuthProvider.getCredential(user?.email.toString(), password.toString())
                    user?.reauthenticate(authCredential)?.addOnSuccessListener {
                        user?.updatePassword(new_password?.text.toString())?.addOnSuccessListener {
                            Toast.makeText(this@ChangePassword, "Password Updated...", Toast.LENGTH_SHORT).show()
                            val intent = Intent()
                            intent.setClass(this@ChangePassword, MainActivity::class.java)
                            startActivity(intent)
                        }?.addOnFailureListener { e -> Toast.makeText(this@ChangePassword, e.message, Toast.LENGTH_SHORT).show() }
                    }
                } else {
                    Toast.makeText(this@ChangePassword, "Input Error...", Toast.LENGTH_SHORT).show()
                }
            }
            else{
                Toast.makeText(this@ChangePassword, "Please enter your password...", Toast.LENGTH_SHORT).show()
            }
        })
    }
}